pyrefly/lib/state/lsp.rs

```diff
@@ -2974,6 +3109,203 @@ impl<'a> Transaction<'a> {
     }
 }
 
+#[derive(Clone, Debug)]
+struct IdentifierRef {
+    name: String,
+    position: TextSize,
+    synthetic_load: bool,
+}
+
+fn contains_range(outer: TextRange, inner: TextRange) -> bool {
+    outer.start() <= inner.start() && outer.end() >= inner.end()
+}
+
+fn ranges_overlap(a: TextRange, b: TextRange) -> bool {
+    a.start() < b.end() && b.start() < a.end()
+}
```

Comment:
you should be able to use TextRange::contains_inclusive and TextRange::overlap for these